src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/POPUserMetricsService.kt

```diff
@@ -1,30 +1,98 @@
   }
 
-  fun recordProbationUsersMetrics(): Int {
-    val otpList = prisonOTPService.getAllOTPs()
-    val cnt = otpList?.size ?: 0
-    registry.gauge(
-      "total_pack_print_count",
-      cnt,
-    )
-    return cnt
+  fun recordProbationUsersLicenceConditionMetrics() {
+    log.info("Started running scheduled POP User metrics job - LicenceCondition")
+    val popUserList = popUserOTPService.getAllOTPs()
+
+    var totalPopUser = 0
+    if (popUserList != null) {
+      totalPopUser = popUserList.size
+    }
+    val prisonList = prisonRegisterApiService.getActivePrisonsList()
+    if (popUserList != null) {
+      var percentStdLicenceCondition: Int
+      var percentOtherLicenceCondition: Int
+      for (prison in prisonList) {
+        var stdLicenceConditionCount = 0
+        var otherLicenceConditionCount = 0
+
+        try {
+          for (popUser in popUserList) {
+            if (popUser.prisoner.prisonId.equals(prison.id)) {
+              val licencesConditions = licenceConditionService.getLicenceConditionsByNomsId(popUser.prisoner.nomsId)
+              val stdLicenceConditionList = licencesConditions?.standardLicenceConditions
+              val otherLicenceConditionList = licencesConditions?.otherLicenseConditions
+              if (!stdLicenceConditionList.isNullOrEmpty()) {
+                stdLicenceConditionCount += 1
+              }
+              if (!otherLicenceConditionList.isNullOrEmpty()) {
+                otherLicenceConditionCount += 1
+              }
+            }
+          }
+          percentStdLicenceCondition = if (stdLicenceConditionCount > 0) {
+            100 - ((stdLicenceConditionCount / totalPopUser) * 100)
+          } else {
+            100
+          }
+
+          percentOtherLicenceCondition = if (otherLicenceConditionCount > 0) {
+            100 - ((otherLicenceConditionCount / totalPopUser) * 100)
+          } else {
+            100
+          }
+
+          val metrics = listOf(
+            PopUserCountMetric(LicenceTag.STANDARD, percentStdLicenceCondition),
+            PopUserCountMetric(LicenceTag.OTHERS, percentOtherLicenceCondition),
+          )
+          popUserCountMetrics.metrics[prison] = metrics
+          val prisonTag = Tags.of("prison", prison.name)
+
+          popUserCountMetrics.metrics[prison]?.forEachIndexed { i, metric ->
+            registry.gauge(
+              "missing_licence_conditions_percentage",
+              prisonTag
+                .and("licenceType", metric.licenceType.label),
+              popUserCountMetrics,
+            ) {
+              it.metrics[prison]?.get(i)?.value?.toDouble()
+                ?: throw RuntimeException("Can't find value for metric $metric. This is likely a coding error!")
+            }
+          }
+        } catch (_: ResourceNotFoundException) {
+        } catch (ex: Exception) {
+          log.warn("Error collecting metrics for popUser Missing Licence Conditions ${prison.name}", ex)
+        }
```

Comment:
would be better to be consistent, one exception is thrown, another is empty statement and the last logs a warning, I'd say either throw all of them or log all of them

Reply:
For consistency this is how it is in Metric Service used for PSFR, so  keeping it as it is!